rfc/expr-context.md

```diff
@@ -0,0 +1,134 @@
+# Expression Context
+
+Cosmo Router provides a context object to an expr-lang VM that can be used to access information about the request, response, and other relevant information. The context object is available at all times during the request's lifecycle.
+
+## request
+
+The client request object is a read-only object that provides information about the incoming request. It is available at all times during the request's lifecycle.
+
+```
+request.id
+request.header
+request.method
+request.body.query
+request.body.operation_name
+request.body.variables
+request.body.extensions
+request.uri.host
+request.uri.path
+request.uri.port
+
+request.client.name
+request.client.version
+request.client.ip
+request.routerConfigVersion
+```
+
+# request.context
+
+The context object provides information about the context of the request. A generic object that can be used to store any information that is relevant to the request.
+
+```
```

Comment:
I'd interpret context as an intermediate datastructure to make handling easier. Agent is a good use case if is more than the raw header value.